EKF/covariance.cpp

```diff
@@ -95,14 +95,7 @@ void Ekf::initialiseCovariance()
 	// record IMU bias state covariance reset time - used to prevent resets being performed too often
 	_last_imu_bias_cov_reset_us = _imu_sample_delayed.time_us;
 
-	// earth frame and body frame magnetic field
-	// set to observation variance
-	for (uint8_t index = 16; index <= 21; index ++) {
-		P(index,index) = sq(_params.mag_noise);
-	}
-
-	// save covariance data for re-use when auto-switching between heading and 3-axis fusion
-	saveMagCovData();
```

Comment:
We will stop saving the mag covariance data at this point in case "_control_status.flags.mag_3D=true".

Reply:
Yes, and this is perfectly fine because we automatically save the mag covariance when we switch out of 3d fusion; there is no need to save it when we are currently using it.